### ISSUE TITLE
New Rule: `n/no-path-concat` for `import.meta.*`

### ISSUE THREAD

**Issue by voxpelli** [eslint-community/eslint-plugin-n#295]
### Rule details

`n/no-path-concat` helps out with how one should handle `__dirname` and `__filename`, but in ESM modules those does not exist (as mentioned in eg. #164) but instead there are three others with similar, yet different, concerns:

* `import.meta.dirname`
* `import.meta.filename`
* `import.meta.url`

### What type of rule is this?

Warns about a potential problem

### Example code

```js
const abc = import.meta.dirname + '/foo/bar.js';
```

### Participation

- [X] I am willing to submit a pull request to implement this rule.

### Additional comments

Either `n/no-path-concat` is extended with those three or a new rule is introduced. I think `n/no-path-concat` could probably be extended? (Hence why I opened #294, because I would have suggested this as a rule change if I could)
